Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift

```diff
@@ -256,7 +256,24 @@ class MessageListVC<ExtraData: ExtraDataTypes>: _ViewController, UICollectionVie
             layoutOptions: layoutOptions,
             for: indexPath
         )
-        
+
+        cell.delegate = .init(
+            didTapOnErrorIndicator: { [weak self] in
+                self?.handleTapOnErrorIndicator(forCellAt: indexPath)
+            },
+            didTapOnThread: { [weak self] in
+                self?.handleTapOnThread(forCellAt: indexPath)
+            },
+            didTapOnAttachment: { [weak self] in
+                self?.handleTapOnAttachment($0, forCellAt: indexPath)
+            },
+            didTapOnAttachmentAction: { [weak self] attachment, action in
+                self?.handleTapOnAttachmentAction(action, for: attachment, forCellAt: indexPath)
+            },
+            didTapOnQuotedMessage: { [weak self] in
+                self?.handleTapOnQuotedMessage($0, forCellAt: indexPath)
+            }
+        )
```

Comment:
I think we can provide some kind of `setDelegate` method that will wrap this boilerplate

Reply:
haha, that is not that straightforward. I find it more flexible to work with `indexPath` instead of `cell/message` but the index path is a context variable which we capture via action closures. We can have a dedicated helper on VC level but on cell level it might look weired